web/source/kmwembedded.ts

```diff
@@ -402,7 +404,7 @@ namespace com.keyman.text {
     }
 
     // Clear any pending (non-popup) key
-    osk.vkbd.keyPending = null;
+    keyman.osk.vkbd.keyPending = null;
```

Comment:
Here you use `keyman.osk` but on line 258 you reference a type hanging under `com.keyman.osk` and on line 357 you use `keymanweb.osk` (from "file closure"-local `keymanweb = window['keyman']`). Any reason for the differences? It would be good to be consistent!

Related: I think we are hoping to reduce the different global names. Are `keyman`, `com.keyman`, `keymanweb` and `KeymanWeb` all globals on `window`? Any others I've forgotten? We should pick one and make sure all changes to code use only that one, even if it takes us longer to refactor the rest of the usages...

Reply:
`kmwembedded.ts` has long been... a bit of a mess in this regard.

It's still not fully reworked from the KMW 2 days, where it always depended on file-global references for the main KeymanWeb objects:

https://github.com/keymanapp/keyman/blob/049ffc90807d738a49931254545f42fe726d759d/web/source/kmwembedded.ts#L125-L128

`keymanweb` is defined as part of the IIFE above and is file-specific.  The others are all globals on window:
- `keyman` - the core KeymanWeb object
- `KeymanWeb` - the keyboard-API interface object (what keyboards look for)
- `com`, `com.keyman` - TS namespacing for KeymanWeb

We explicitly made the decision to split `keyman` and `KeymanWeb` back in 10.0, as I recall.  That way, keyboards don't have access to _all_ the KMW internals.  Both those names existed back in 2.0, but were aliases for each other then.

> We should pick one and make sure all changes to code use only that one, even if it takes us longer to refactor the rest of the usages...

Pretty sure that'd be one of the benefits to the `npm` distribution roadmap goal - modular JS is much better at avoiding this kind of pollution.

...

That said... after some research, there may _actually_ be a way to sandbox our keyboards, allowing us to drop the `KeymanWeb` global.

https://stackoverflow.com/a/61203013 shows a way we could intercept `<script>` tag loading for any such tags KMW itself causes, like with `setActiveKeyboard`.  We could then IIFE the script and define the previously-global `KeymanWeb` within that.  Which I'm now totally on board with, whenever we can find the time to invest in that direction.

Reply:
Though... good catch on this point.  That line does need to be fixed.

Reply:
Not so sure about that stackoverflow solution ... feels a bit strings and rubber bands.

Reply:
> That said... after some research, there may _actually_ be a way to sandbox our keyboards, allowing us to drop the `KeymanWeb` global.

After further thought, we need to keep the `KeymanWeb` global. We don't always control how keyboard scripts are loaded; they could well be embedded inline (think a webpack-style optimised load). Clever tricks like that SO option tend to break otherwise-normal usages.

So, from my perspective, we deprecate `com`, `com.keyman` etc and have `KeymanWeb` and `keyman` globals only. Is that correct?

Reply:
Once we're able to do so, correct.  It's going to be a longhaul toward that... but at least we never even _published_ any official documentation toward `com` or `com.keyman`.